src/coord/src/coord.rs

```diff
@@ -3035,6 +3048,10 @@ impl Coordinator {
     // Notify the timestamper thread that a source has been created or dropped.
     async fn update_timestamper(&mut self, source_id: GlobalId, create: bool) {
         if create {
+            let bindings = self
```

Comment:
Should this happen after the entry.item so we don't do unnecessary work if it doesn't end up being used?

Reply:
there was some annoying lifetime reason i had to move it up here. this function is only called in 2 or 3 places where I think its basically provable that it will only get called with sources so i think its ok